src/System.Buffers.Experimental/System/Buffers/BufferExtensions.cs

```diff
@@ -218,7 +218,117 @@ private static unsafe int IndexOfVectorized(ref byte searchSpace, byte value, in
         exitFixed:;
                 return offset;
             }
+        }
+
+        public static bool TryIndicesOf(this Span<byte> buffer, byte value, Span<int> indices, out int numberOfIndices)
+        {
+            var length = buffer.Length;
+            if (length == 0 || indices.Length == 0)
+            {
+                numberOfIndices = 0;
+                return false;
+            }
+
+            return TryIndicesOf(ref buffer.DangerousGetPinnableReference(), value, length, indices, out numberOfIndices);
+        }
+
+        public static bool TryIndicesOf(this ReadOnlySpan<byte> buffer, byte value, Span<int> indices, out int numberOfIndices)
```

Comment:
Yeah, it would be great if you could validate/test this new API in the context of the following routine: https://github.com/aspnet/KestrelHttpServer/blob/davidfowl/changes/src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/Frame.cs#L1492